python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -104,6 +104,83 @@ def dist_inference(self, g, get_input_embeds, batch_size, fanout,
         return dist_inference(g, self, get_input_embeds, batch_size, fanout,
                               edge_mask=edge_mask, task_tracker=task_tracker)
 
+
+def dist_minibatch_inference(g, gnn_encoder, get_input_embeds, batch_size, fanout,
```

Comment:
why do we want to have mini-batch inference on the entire graph?

Reply:
If there is a custom model, we may not be able to do layer-by-layer full graph inference.
That's why we need mini-batch inference.

Reply:
i see.